packages/studio/src/main.tsx

```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
-import { App, hotReloadStore } from "@yext/studio-ui";
-import { StudioHMRPayload, StudioHMRUpdateID } from "@yext/studio-plugin";
```

Comment:
Out of curiosity, what does importing `StudioHMRUpdateID` from `studio-ui` do in terms of performance?

Reply:
it's a type import so nothing! I tested both ways and specifying it as a type import with the `type` keyword doesn't affect whether or not network requests are made!

If it we were importing say Typeguards instead (i.e. just any non-type) vite would make network requests for every file under `packages/studio-plugin/src/index.ts`. 
This morning (because work is unfortunately somewhat ingrained in my mind) I was coincidentally reading up on vite prebundling, and this behavior can be explained by how vite treats linked packages in a monorepo as source code by default https://vitejs.dev/guide/dep-pre-bundling.html
Pre bundling can be turned on by adding `optimizeDeps.include` for the specific packages you want, which would address this behavior as well. 

I **think** we still want this off though because bundling studio-ui as a separate build step is better than having vite prebundle take time prebundling things before the dev server launches. Pre bundling still takes some amount of time (I haven't tested but I assume it would take the same amount of time as the current build for studio-ui takes which for me is about 10 seconds) and we also have less visibility about the bundle size when we rely on prebundling.

We could turn on prebundling on TOP of bundling ourselves, as a kind of safety net. But it's completely unnecessary and I'm worried would only mask performance issues

Reply:
I misread your comment, I thought you were asking about StudioHMRPayload not StudioHMRUpdateID.
The behavior for StudioHMRUpdateID is the same as described above for any non-type import